File: my-tech/src/representations/Product.js
export default class Product{
    constructor(id, name, provider, price, stock){
        this.id = id;
        this.name = name;
        this.provider = provider;
        this.price = price;
        this.stock = stock;
    }

    static fromStateProduct(product){
        return new Product(product.id, product.name, product.provider, product.price, product.stock);
    }

    static fromApiResponseProduct(response){
        return new Product(response.id, response.name, response.provider, response.price, response.stock);
    }
}

File: MyTech.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTech.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Provider { get; set; }
        public double Price { get; set; }
        public int Stock { get; set; }

        public Product()
        {

        }

        public Product(int id, string name, string provider, double price, int stock)
        {
            Id = id;
            Name = name;
            Provider = provider;
            Price = price;
            Stock = stock;
        }
    }
}
